cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/etl/api/PipelineConfigurer.java

```diff
@@ -31,4 +33,18 @@
    * @return stage configurer
    */
   StageConfigurer getStageConfigurer();
+
+  /**
+   * @return the engine for this pipeline
+   */
+  Engine getEngine();
+
+  /**
+   * Set pipeline properties that will be applied to each run of the pipeline.
+   * Depending on the engine used, the properties will be added to the SparkConf for each run or the mapreduce
```

Comment:
oh sorry I thought this was somewhere else. When set programmatically, no prefixing is needed. The prefixing is only used when somebody sets the properties through the application config.